zipline/pipeline/mixins.py

```diff
@@ -109,7 +111,16 @@ def _compute(self, windows, dates, assets, mask):
         compute = self.compute
         missing_value = self.missing_value
         params = self.params
-        out = full_like(mask, missing_value, dtype=self.dtype)
+        outputs = self.outputs
+        if outputs:
+            out = recarray(
+                mask.shape,
+                formats=[self.dtype.str] * len(outputs),
+                names=outputs,
+            )
+            out[:] = array([missing_value])
```

Comment:
Do we need to cast to an array here?  It seems to work to just assign the missing value directly:

```
In [2]: recarray((2, 2), formats=['i', 'i'], names=['a', 'b'])
Out[2]:
rec.array([[(-396702552, 32663), (-396702552, 32663)],
       [(-392389200, 32663), (-392592440, 32663)]],
      dtype=[('a', '<i4'), ('b', '<i4')])

In [3]: r = recarray((2, 2), formats=['i', 'i'], names=['a', 'b'])

In [4]: r[:] = 0

In [5]: r
Out[5]:
rec.array([[(0, 0), (0, 0)],
       [(0, 0), (0, 0)]],
      dtype=[('a', '<i4'), ('b', '<i4')])
```